src/app/views/groups/edit/EditGroup.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from "react";
+import { withRouter } from "react-router";
+import PropTypes from "prop-types";
+import BackButton from "../../../components/back-button";
+import Loader from "../../../components/loader";
+import Input from "../../../components/Input";
+import validate from "./validate";
+import isEqual from "lodash/isEqual";
+import isEmpty from "lodash/isEmpty";
+import FormFooter from "../../../components/form-footer/FormFooter";
+
+const EditGroup = props => {
+    const id = props.params.id;
+
+    useEffect(() => {
+        if (id) {
+            props.loadGroup(id);
+        }
+    }, [id]);
+
+    const { group, loading, updateGroup } = props;
+    const [values, setValues] = useState(null);
+    const [errors, setErrors] = useState({});
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const hasErrors = !isEmpty(errors);
+    const hasNewValues = !isEqual(values, group);
+    const specialGroup = group?.precedence === (1 || 2);
+
+    const routerWillLeave = nextLocation => {
+        if (hasNewValues && !isSubmitting) return "Your work is not saved! Are you sure you want to leave?";
+    };
+
+    useEffect(() => {
+        props.router.setRouteLeaveHook(props.route, routerWillLeave);
+    });
+
+    useEffect(() => {
+        if (group) {
+            setValues({ ...group }); // ths is not necessary but in future we will edit more fields
+        }
+    }, [group]);
+
+    const handleSubmit = e => {
+        if (e) e.preventDefault();
+        setErrors(validate(values));
+        setIsSubmitting(true);
+
+        if (hasErrors || values.name === "") return;
+        updateGroup(id, { name: values.name, precedence: 10 });
+    };
+
+    const handleChange = e => {
+        const key = e.target.name;
+        const val = e.target.value;
+        setIsSubmitting(false);
+
+        if (hasErrors) {
+            delete errors[e.target.name];
+            setErrors(errors);
+        }
+        setValues(values => ({ ...values, [key]: val }));
```

Comment:
Just a question: On line 22 you have `const [values, setValues] = useState(null);`. Are you defaulting or mutating `values` to an empty array somewhere else, you might get a not iterable TypeError here?